File: LeetCode/C#/400-499/455.cs
using System;

namespace LeetCode455
{
    public class Solution {
        public int FindContentChildren(int[] g, int[] s) {
            Array.Sort(g); 
            Array.Sort(s);
            int child = 0;
            int cookie = 0;
            while(child < g.Length && cookie < s.Length){
                if(g[child] <= s[cookie]){
                    child++;
                }
                cookie++;
            }
            
            return child;
        }
    }
    class Program
    {
        static void Main(string[] args)
        {
            int[] g = new int[] {1, 2};
            int[] s = new int[] {1, 2, 3};
            Console.WriteLine(new Solution().FindContentChildren(g, s));
            Console.Read();
        }
    }
}

File: LeetCode/C++/400-499/455.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        int child = 0;
        int cookie = 0;
        while(child < g.size() && cookie < s.size()){
            if(g[child] <= s[cookie]){
                child++;
            }
            cookie++;
        }
        
        return child;
    }
};


int main()
{
	vector<int> g{1,2,3}; 
	vector<int> s{1,2}; 
	Solution sol;
	int ans = sol.findContentChildren(g, s);
	
    cout << ans << endl;

    return 0;
}
